### PR TITLE
Airdrop claim

### PR DESCRIPTION
**Pull request by amityadav0**
## Description

Closes:
* https://github.com/elys-network/issues/issues/XXX
* https://github.com/elys-network/issues/issues/XXX

### What has Changed?

What specific problem were you aiming to address, and how did you successfully resolve it? If tests were not uploaded for this pull request or if coverage decreased, please provide an explanation for the change.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/elys-network/elys/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/elys-network/elys/blob/main/docs/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/elys-network/elys/blob/main/CONTRIBUTING.md#testing)
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### Deployment Notes

Are there any specific considerations to take into account when deploying these changes? This may include new dependencies, scripts that need to be executed, or any aspects that can only be evaluated in a deployed environment.

### Screenshots and Videos

Please provide any relevant before and after screenshots by uploading them here. Additionally, demo videos can be highly beneficial in demonstrating the process.

### CODE REVIEW

**Review by cosmic-vagabond — APPROVED** [review]

**Comment by amityadav0** [comment]
Please don't merge yet, need to add a list of accounts which were missed

File: proto/elys/commitment/tx.proto [hunk 1]
```diff
@@ -204,3 +206,22 @@ message MsgUnstakeResponse {
   uint64 code = 1;
   string result = 2;
 }
+
+message MsgClaimAirdrop {
+  option (cosmos.msg.v1.signer) = "creator";
+  option (amino.name) = "commitment/MsgClaimAirdrop";
+  string creator = 1 [ (cosmos_proto.scalar) = "cosmos.AddressString" ];
```
[the remark this answers is not in the document]
Comment (amityadav0) [reply proto/elys/commitment/tx.proto]: done

File: x/commitment/keeper/airdrop.go [hunk 2]
```diff
@@ -151,3 +154,62 @@ func (k Keeper) GetAllCadets(ctx sdk.Context) (list []*types.Cadet) {
...
+
+func (k Keeper) GetAirdropClaimed(ctx sdk.Context, address sdk.AccAddress) (val types.AirdropClaimed) {
+	store := runtime.KVStoreAdapter(k.storeService.OpenKVStore(ctx))
+
+	b := store.Get(types.GetAirdropClaimedKey(address))
+
+	if b != nil {
+		k.cdc.MustUnmarshal(b, &val)
+	} else {
+		val.Claimed = false
```
[the remark this answers is not in the document]
Comment (amityadav0) [reply x/commitment/keeper/airdrop.go:166]: done

File: proto/elys/commitment/tx.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (amityadav0) [reply proto/elys/commitment/tx.proto:213]: It stopped working, so reverted it

`raw_log: field creator not found in message elys.commitment.MsgClaimAirdrop`

File: x/commitment/keeper/msg_server_claim_airdrop.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	"context"
+
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/elys-network/elys/x/commitment/types"
+	ptypes "github.com/elys-network/elys/x/parameter/types"
+)
+
+var AirdropWallet = "elys1wk7jwkqt2h9cnpkst85j9n454e4y8znlgk842n"
```
Comment (avkr003) [inline x/commitment/keeper/msg_server_claim_airdrop.go:13]: This is literal string. Can be made const

File: proto/elys/commitment/tx.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (avkr003) [reply proto/elys/commitment/tx.proto:213]: Update signer value also

File: x/commitment/keeper/msg_server_claim_airdrop.go (hunk 3 above)
Comment (amityadav0) [reply x/commitment/keeper/msg_server_claim_airdrop.go:13]: this is done for running tests, test suite uses cosmos... based address

**Pull request merged**